main/main.cpp

```diff
@@ -2428,6 +2428,19 @@ bool Main::start() {
 
 	OS::get_singleton()->set_main_loop(main_loop);
 
+	uint64_t minimum_time_msec = GLOBAL_DEF("application/boot_splash/minimum_display_time_msec", 0);
+	ProjectSettings::get_singleton()->set_custom_property_info("application/boot_splash/minimum_display_time_msec",
+			PropertyInfo(Variant::INT,
+					"application/boot_splash/minimum_display_time_msec",
+					PROPERTY_HINT_RANGE,
+					"0,100,1,or_greater")); // No negative numbers
+	uint64_t minimum_time = 1000 * minimum_time_msec;
+	uint64_t elapsed_time = OS::get_singleton()->get_ticks_usec();
+	int64_t time_left = minimum_time - elapsed_time;
+	if (time_left > 0) {
+		OS::get_singleton()->delay_usec(time_left);
+	}
+
```

Comment:
Looks fine, personally I'd probably modify the if clause to do:
```
	uint64_t minimum_time = 1000 * minimum_time_msec;
	uint64_t elapsed_time = OS::get_singleton()->get_ticks_usec();
	if (elapsed_time < minimum_time)
	{
		OS::get_singleton()->delay_usec(minimum_time-elapsed_time);
	}
```
just because I'm not entirely clear on what the promotion rules are for this expression:
`int64_t time_left = minimum_time - elapsed_time;`

If the right hand side is evaluated as `uint64_t` before being converted to `int64_t`, and minimum_time is less than elapsed time, you are relying on the rollover overflow behaviour of `uint64_t` being identical to `int64_t` (I think it is bitwise the same and I'm not against this kind of thing in all cases, but this just seems a bit unclear and unnecessarily dodgy in this case, when there is no need for overflow).

In cases of conversion from `uint` to `int` and vice versa I usually end up trying to be explicit about conversion imo rather than implicit, so the intention is clear (I've personally caused a few bugs due to this kind of thing in my time lol :grin:).